framework/src/main/java/org/tron/core/net/messagehandler/SyncBlockChainMsgHandler.java

```diff
@@ -87,6 +87,18 @@ private boolean check(PeerConnection peer, SyncBlockChainMessage msg) throws P2p
 
   private LinkedList<BlockId> getLostBlockIds(List<BlockId> blockIds) throws P2pException {
 
+    BlockId unForkId = getUnForkId(blockIds);
+    LinkedList<BlockId> ids = getBlockIds(unForkId.getNum());
+
+    if (ids.isEmpty() || !unForkId.equals(ids.peekFirst())) {
+      unForkId = getUnForkId(blockIds);
```

Comment:
If I'm not mistaken, is it just a retry?

Reply:
Yes, it can be solved by retrying.